src/main/java/edu/ie3/datamodel/io/sink/InfluxDbSink.java

```diff
@@ -0,0 +1,311 @@
+import edu.ie3.datamodel.io.extractor.NestedEntity;
+import edu.ie3.datamodel.io.processor.ProcessorProvider;
+import edu.ie3.datamodel.io.processor.timeseries.TimeSeriesProcessorKey;
+import edu.ie3.datamodel.models.UniqueEntity;
+import edu.ie3.datamodel.models.input.container.JointGridContainer;
+import edu.ie3.datamodel.models.result.ResultEntity;
+import edu.ie3.datamodel.models.timeseries.TimeSeries;
+import edu.ie3.datamodel.models.timeseries.TimeSeriesEntry;
+import edu.ie3.datamodel.models.value.Value;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.influxdb.InfluxDB;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+
+/** InfluxDB Sink for result and time series data */
+public class InfluxDbSink implements DataSink {
+  private static final Logger log = LogManager.getLogger(InfluxDbSink.class);
+  /** Field name for timestamp field in result entities */
+  private static final String FIELD_NAME_TIMESTAMP = "timestamp";
+  /** Field name for timestamp field in time series */
+  private static final String FIELD_NAME_TIME = "time";
+  /** Field name for input model uuid field in result entities */
+  private static final String FIELD_NAME_INPUT = "inputModel";
+
+  private final InfluxDbConnector connector;
+  private final FileNamingStrategy fileNamingStrategy;
+  private final ProcessorProvider processorProvider;
+
+  /**
+   * Initializes a new InfluxDbWeatherSource
+   *
+   * @param connector needed for database connection
+   * @param fileNamingStrategy needed to create measurement names for entities
+   */
+  public InfluxDbSink(InfluxDbConnector connector, FileNamingStrategy fileNamingStrategy) {
+    this.connector = connector;
+    this.fileNamingStrategy = fileNamingStrategy;
+    this.processorProvider =
+        new ProcessorProvider(
+            ProcessorProvider.allResultEntityProcessors(),
+            ProcessorProvider.allTimeSeriesProcessors());
+  }
+
+  /**
+   * Initializes a new InfluxDbWeatherSource with a default FileNamingStrategy
+   *
+   * @param connector needed for database connection
+   */
+  public InfluxDbSink(InfluxDbConnector connector) {
+    this(connector, new FileNamingStrategy());
+  }
+
+  @Override
+  public void shutdown() {
+    connector.shutdown();
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persist(C entity) {
+    Set<Point> points = extractPoints(entity);
+    writeAll(points);
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persistAll(Collection<C> entities) {
+    Set<Point> points = new HashSet<>();
+    for (C entity : entities) {
+      points.addAll(extractPoints(entity));
+    }
+    writeAll(points);
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persistIgnoreNested(C entity) {
+    try {
+      if (!(entity instanceof ResultEntity))
```

Comment:
Is this straight forward? It would prevent time series from being persisted, doesn't it?

Reply:
I understood the method to be mostly for "regular" results. But maybe those persist(All)IgnoreNested are better to implemented with just an IllegalArgumentException as InfluxDB doesn't support anything except timeseries?